main/acle.md

```diff
@@ -2107,6 +2115,8 @@ The following attributes trigger the multi version code generation:
 * The `default` version means the version of the function that would
   be generated without these attributes.
 * `name` is the dependent features from the tables below.
+  * If a feature depends on an other feature as defined by the Architecture
```

Comment:
This is still a bit vague to me in terms of what happens. In particular it doesn't actually convey that e.g. using `fcmla` will move you to `armv8.3-a` unless you're at `armv8.3-a` or higher already?

Reply:
Trick, ArmARM says: "From Armv8.3, if Advanced SIMD or Floating-point is implemented, the only permitted value is 0b0001." so it is valid to use any feature that is mandatory on Armv8.3.
Those shall not be disabled/trapped by a higher ELs.
So FJCVTZS or RETAA/PAC* would be usable in this context too since those are mandatory and RETAA will work even if PAC is disabled. I'm not sure if I can give a complete matrix here.

Reply:
Well not sure how useful the ArmARM is in this regard, it also says
```
An Armv8.x compliant implementation can include any arbitrary subset of the architectural features of
Armv8.(x+1), subject only to those constraints that require that certain features be implemented together.
```
Which means e.g. technically speaking you should be able to use `fcmla` at `Armv8.2-A` as well, but compilers don't allow this, gating the operations on `Armv8.3-A`.

But what I'm really after here is knowing what needs to be done, if a user compiles with `-march=armv8-a` and then uses `fcmla` FMV attribute, what needs to happen here?

Reply:
> But what I'm really after here is knowing what needs to be done, if a user compiles with -march=armv8-a and then uses fcmla FMV attribute, what needs to happen here?

The user's intention is to use `fcmla` instructions in the function, so I architecturally no other feature need to be enabled to compile the function. That would be the safest way to go.
A counter example would be `sve-aes` where `sve` is required.

Reply:
I'm not sure we can do that in GNU land. Both the compiler and the assembler don't have an individual feature flag for instructions such as `fcmla` that are tied to an architecture.  This is because the internal guidelines for architecture enablement mandated this.

We can retro-actively split these (we've done so in the past with `+crypto`), but that requires alignment from the architecture enablement teams as we'd need to introduce new use visible extension flags. `fcmla` isn't the only feature where this was done FWIW.